app/models/recent.rb

```diff
@@ -9,11 +9,32 @@ class Recent < ActiveRecord::Base
   has_one :user, through: :classification
   has_one :user_group, through: :classification
 
+  belongs_to :project
+  belongs_to :workflow
+  belongs_to :user
+  belongs_to :user_group
+
   validates_presence_of :classification, :subject
 
+  # TODO: modify the following validation to run all the time
+  # once the recents schema has been migrated from has_one through associations
+  validates_presence_of :project_id, :workflow_id, :user_id, only: :create
+
+  before_validation :copy_classification_fkeys
+
   def self.create_from_classification(classification)
-    classification.subject_ids.each do |sid|
-      create!(subject_id: sid, classification: classification)
+    classification.subject_ids.map do |sid|
```

Comment:
Maybe just call that block parameter `subject_id`, to make sure any grepping will find it.